### ISSUE TITLE
CodeQwen returns extra white space for code completion

### ISSUE THREAD

**Issue by ycclnn** [TabbyML/tabby#1947]
![image](https://github.com/TabbyML/tabby/assets/56081290/538a7574-6a90-4600-8b1f-04581e5879cc)

Other models like deepseek works without a problem.

**Comment by wsxiaoys** [comment]
I have also observed the issue and am looking into the debug process. It seems that the tokenizer treats `_` as a space, regardless of the context for CodeQwen.

**Comment by wsxiaoys** [comment]
Can confirm this presents in upstream llama.cpp as well:

cross posted at: https://github.com/ggerganov/llama.cpp/issues/7050

**Comment by ycclnn** [comment]
> Can confirm this presents in upstream llama.cpp as well:
> 
> cross posted at: [ggerganov/llama.cpp#7050](https://github.com/ggerganov/llama.cpp/issues/7050)

yeah, white space exists for using vllm so its a model thing rather than serving framework thing. I was thinking shiftleft the prefix cursor a bit and then check the completion result to overcome this and thats the only workaround in my mind.

**Comment by ycclnn** [comment]
Observed that codeqwen not always returns extra white space, and sometimes the white space is meaningful. Therefore, trim the leading white space may not be an approach.
